### PR TITLE
[CIVIS-1824] ENH add script to check if Civis API spec has updated

### PR DESCRIPTION
**Pull request by jacksonlee-civis**
This pull request add the script `tools/check_if_civis_api_spec_has_updated.py` for checking if the Civis API spec is up-to-date. This script is scheduled to run on Civis Platform and notifies civis-python maintainers if updates are detected.

---

- [x] (For Civis employees only) Reference to a relevant ticket in the pull request title
- [x] Changelog entry added to `CHANGELOG.md` at the repo's root level
- [x] Description of change in the pull request description
- [x] If applicable, unit tests have been added and/or updated
- [x] The CircleCI builds have all passed

### CODE REVIEW

File: tools/update_civis_api_spec_json.py
Comment (jacksonlee-civis) [inline tools/update_civis_api_spec_json.py:1]: This download code is now the function `download_latest_api_spec()` in `src/civis/resources/_api_spec.py` above.

File: tools/update_client_pyi.py
Comment (jacksonlee-civis) [inline tools/update_client_pyi.py:1]: Script not needed anymore, as `client.pyi` is updated whenever `civis_api_spec.json` is updated by `tools/update_civis_api_spec.py`.

File: tools/check_if_civis_api_spec_has_updated.py
Comment (jacksonlee-civis) [inline tools/check_if_civis_api_spec_has_updated.py:1]: Civis Platform script URL in the linked internal ticket

File: src/civis/resources/_resources.py [hunk 1]
```diff
@@ -17,20 +17,16 @@
...
 # civis_api_spec.json can be updated
-# by running the tools/update_civis_api_spec_json.py script.
+# by running the tools/update_civis_api_spec.py script.
 API_SPEC_PATH = os.path.join(
     os.path.dirname(os.path.realpath(__file__)),
     "civis_api_spec.json",
 )
 with open(API_SPEC_PATH) as f:
     API_SPEC = json.load(f, object_pairs_hook=OrderedDict)
-BASE_RESOURCES_V1 = sorted(
```
Comment (jacksonlee-civis) [inline src/civis/resources/_resources.py:27]: `BASE_RESOURCES_V1` needs to be removed, as it controlled what API resources would be included / excluded by `civis.resources.generate_classes_maybe_cached`. The resources included in `BASE_RESOURCES_V1` were based on the versioned `civis_api_spec.json`, which would directly interfere with the goal of the new script `tools/check_if_civis_api_spec_has_updated.py`.

File: src/civis/resources/_resources.py [hunk 2]
```diff
@@ -51,11 +47,12 @@
...
 
 def exclude_resource(path, api_version):
-    if api_version == "1.0":
-        include = any(path.startswith(x) for x in BASE_RESOURCES_V1)
-    else:
-        include = True
-    return not include
+    # TODO: api_version is not used here.
+    #   Dropping it would affect upstream code, including civis.APIClient.
+    #   We may need to deprecate api_version in the future.
```
Comment (jacksonlee-civis) [inline src/civis/resources/_resources.py:52]: Deprecating the `api_version` param across the board, if done at all, would come from a future PR.

**Review by Drew-Pappas — APPROVED** [review]
LGTM! Validated the internal ticket + script

**Pull request merged**